### PR TITLE
[Bug,Enhancement] Fix and improve Starter selector UI

### PR DESCRIPTION
**Pull request by KimJeongSun**
## What are the changes?

This PR enhances the hybrid filters with hovering functionality.
1. The dropdowns of all hybrid filters no longer obscure the Pokémon list.
2. The default setting for hybrid filters is ALL OFF. (In classic mode, the Gen filter defaults to Gen 1 with the cursor hovering over it.)

## Why am I doing these changes?

1. Hybrid filters change the filtered list in real-time with cursor movement, and it is much more useful to not obscure the Pokémon list in hovering mode.
2. The default setting for all hybrid filters should be hovering, as multi-selecting when necessary is the standard usage. Keeping the default settings for type and dex as all selected, as before, is inconvenient since most users will need to clear all selections before using filters effectively.

## What did change?

1. Changed all hybrid windows to appear to the left of the Pokémon list. This adjustment ensures that any additional hybrid filters added in the future will automatically be positioned to the left.
2. Refactored the way default settings are stored. The previous method could not remember the active cursor position. This allows the resetFilter function to be more intuitive and concise. 
 -> This fixes the bug where the Gen ALL hover mode wasn't saved when moving filters, as seen in the before video.

### Screenshots/Videos

before

https://github.com/user-attachments/assets/24f15b69-f6b5-4651-b2ae-6a0b25241545

after

https://github.com/user-attachments/assets/ce1152a2-ebca-4f8b-9ba6-701463336080

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by MokaStitcher — CHANGES_REQUESTED** [review]
Some of the functionality got broken with the Sort filter:

The category doesn't display as changed if the direction is the default one, even if a different sort than the default one was selected

Resetting does not work properly if you set the sorting direction to descending before exiting and creating a new game.
Steps to reproduce:
- go to sort filter
- select Cost, descending, this requires two inputs, one to switch to Cost, the other to switch to descending
- exit out of starter select
- start a new game
- go to sort filter
- select Cost, it will directly be set to descending order
This is because of the way all sorting options keep track of the current sorting direction and these values don't get reset anymore with your changes.

File: src/ui/starter-select-ui-handler.ts [hunk 1]
```diff
@@ -914,19 +914,12 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
    */
   resetFilters() : void {
     const genDropDown: DropDown = this.filterBar.getFilter(DropDownColumn.GEN);
-    if (this.scene.gameMode.isChallenge) {
-      // In challenge mode all gens are selected by default
-      genDropDown.defaultCursor = 0;
```
[the remark this answers is not in the document]
Comment (KimJeongSun) [reply src/ui/starter-select-ui-handler.ts:919]: current, it is using in dropdown.ts for update default cursor when not hovering mode. so let's keep it

File: src/ui/dropdown.ts [hunk 2]
```diff
@@ -457,24 +459,35 @@ export class DropDown extends Phaser.GameObjects.Container {
...
+    return settings;
+  }
+
   /**
    * Check whether the values of all options are the same as the default ones
    * @returns true if they are the same, false otherwise
    */
   public hasDefaultValues(): boolean {
-    const currentValues = this.getVals();
+    const currentValues = this.getSettings();
```
[the remark this answers is not in the document]
Comment (KimJeongSun) [reply src/ui/dropdown.ts:479]: I gave it a try, but it doesn't seem to work properly as the dir and cursor parts seem to differ by type. Let's plan to refactor it next time.

[the remark this answers is not in the document]
Comment (KimJeongSun) [reply src/ui/dropdown.ts:480]: I will try refactoring this function, after that i will make it ready to review again

[the remark this answers is not in the document]
Comment (KimJeongSun) [reply src/ui/dropdown.ts:480]: updated

**Review by MokaStitcher — APPROVED** [review]
looks good!

**Pull request merged**